src/audio/output.rs

```diff
@@ -0,0 +1,35 @@
+//! Provides structures and functions used to get audio outputs.
+
+// We have to use types from this to provide an output iterator type.
+extern crate cpal;
+
+use rodio::Endpoint;
+use self::cpal::EndpointsIterator;
+use rodio::*;
+
+use std::iter::Map;
+
+/// A speaker(s) through which audio can be played.
+pub struct Output {
+    pub(crate) endpoint: Endpoint,
+}
+
+impl Output {
+    /// Gets the name of the output
+    pub fn get_name(&self) -> String {
+        self.endpoint.get_name()
+    }
+}
+
+/// An iterator over outputs
+pub type OutputIterator = Map<EndpointsIterator, fn(Endpoint) -> Output>;
```

Comment:
Could you just create a struct which owns `EndpointIterator` and map it in the `next` method? Would probably be more efficient.